app/views/contexts/calculated_summary_context.py

```diff
@@ -24,7 +25,9 @@ class CalculatedSummaryContext(Context):
     def build_groups_for_section(
         self, section: Mapping, return_to_block_id: str, current_location: Location
     ) -> List[Group]:
-        routing_path = self._router.routing_path(section["id"])
+        routing_path = self._router.routing_path(
+            section["id"], list_item_id=current_location.list_item_id
+        )
```

Comment:
I know this is existing but is this evaluation even needed?. Although the method is build groups for section, this is only every called for the current locations section, so we don't need this re-evaluation?. You should just be able to use `self._routing_path`?

Reply:
I have replaced this with self._routing_path now 👍